src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1052 @@
+
+with this bash script you can overload your scala play framework application running in the pod, and see how the
+kubernetes auto-scale the application in response to the incoming load. Adjust the frequencies per minutes and the duration
+of the overload in minutes.
+
+```bash
+#!/bin/bash
+
+# Configuration variables.
+# Memory workload configs
+#d: duration of the memory hold(seconds)
+#m: total memory to hold(MB)
+#s: memory chunks to allocate progressively(MB)
+#l: flag to enable logging (default: true)
+
+ENDPOINT="http://k8splayworkload.com/createMemoryLoad?d=180&m=512&s=28&l=true"  # Target MEMORY WORKLOAD HTTP endpoint
+# CPU workload config: n is the order of fibonacci to compute
+#ENDPOINT="http://k8splayworkload.com/fibonacci?n=50"  # Target CPU WORKLOAD HTTP endpoint
+CALLS_PER_MINUTE=70                # Number of calls per minute
+TOTAL_MINUTES=3                    # Total duration in minutes
+
+# Display configuration
+echo "Starting HTTP load generator with the following configuration:"
+echo "Endpoint: $ENDPOINT"
+echo "Frequency: $CALLS_PER_MINUTE calls per minute"
+echo "Duration: $TOTAL_MINUTES minutes"
+echo "Total calls to be made: $((CALLS_PER_MINUTE * TOTAL_MINUTES))"
+echo "----------------------------------------"
+
+# Calculate timing
+SLEEP_TIME=$(bc <<< "scale=6; 60 / $CALLS_PER_MINUTE")
+TOTAL_SECONDS=$((TOTAL_MINUTES * 60))
+START_TIME=$(date +%s)
+END_TIME=$((START_TIME + TOTAL_SECONDS))
+
+# Initialize counter
+COUNTER=0
+
+# Run the load test
+while [ $(date +%s) -lt $END_TIME ]; do
+  # Make HTTP request without waiting for response
+  curl -s -o /dev/null -w "" "$ENDPOINT" &
+
+  # Increment counter
+  COUNTER=$((COUNTER + 1))
+
+  # Display progress every 10 requests
+  if [ $((COUNTER % 10)) -eq 0 ]; then
+    ELAPSED_TIME=$(($(date +%s) - START_TIME))
+    ELAPSED_MINUTES=$(bc <<< "scale=2; $ELAPSED_TIME / 60")
+    PERCENT_COMPLETE=$(bc <<< "scale=2; ($ELAPSED_TIME / $TOTAL_SECONDS) * 100")
+    echo "Progress: $COUNTER requests sent ($ELAPSED_MINUTES minutes elapsed, $PERCENT_COMPLETE% complete)"
+  fi
+
+  # Sleep for calculated time
+  sleep $SLEEP_TIME
+done
+
+# Final report
+TOTAL_TIME=$(($(date +%s) - START_TIME))
+ACTUAL_RATE=$(bc <<< "scale=2; $COUNTER / ($TOTAL_TIME / 60)")
+
+echo "----------------------------------------"
+echo "Load test completed:"
+echo "Total requests sent: $COUNTER"
+echo "Actual duration: $(bc <<< "scale=2; $TOTAL_TIME / 60") minutes"
+echo "Average rate: $ACTUAL_RATE requests per minute"
+```
+Having an eye on the kubernetes dashboard page you can see how the cluster reacts in response to the
+load. You'll notice how it scales by adding the number of pods and later when the overload is passed it
+scales down.
+
+#### Zero down-time deployment
+We have defined a zero-downtime deployment in the attribute `stragey`. to see how this works we can change the actual version
```

Comment:
- `strategy`
- capitalize start of sentences

Reply:
Ok Got it